test/extensions/filters/http/wasm/wasm_filter_integration_test.cc

```diff
@@ -114,12 +129,21 @@ class WasmFilterIntegrationTest
     ASSERT_TRUE(fake_upstream_connection_->close());
     ASSERT_TRUE(fake_upstream_connection_->waitForDisconnect());
   }
+
+  static std::string
+  testParamsToString(const ::testing::TestParamInfo<
+                     std::tuple<std::tuple<std::string, std::string, bool>, Http::CodecType>>& p) {
+    return fmt::format("{}_{}_{}_{}", std::get<2>(std::get<0>(p.param)) ? "downstream" : "upstream",
```

Comment:
optional: if you decide to stick with nested tuples for test params, structured bindings might make this a little easier to read, e.g.
```
auto [wasm_test_params, codec] = p.param;
auto [runtime, language, direction] = wasm_test_params;
return fmt::format("{}_{}_{}_{}", direction ? "downstream" : "upstream", runtime, language, ...);
```